Bottom: 4c1e800808297c82b906eb5c87df9168edb5ee44
Top:    8f66ef44921f1a858d3f7e3a3d8e3f92eaa61734
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-25 16:00:33 +0100

draw.c: Separate "Gathering" routines from real drawing routines


---

diff --git a/src/draw.c b/src/draw.c
index 09e183e..fde0bfd 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -1177,11 +1177,19 @@ DrawElementPackageLowLevel (ElementTypePtr Element)
 void
 DrawVia (PinTypePtr Via)
 {
-  if (!Gathering)
-    SetPVColor (Via, VIA_TYPE);
-  DrawPinOrViaLowLevel (Via, true);
+  assert (Gathering);
+
+  GatherPV (Via);
+  GatherPVName (Via);
+}
+
+static void
+draw_pcb_via (PinTypePtr Via)
+{
+  SetPVColor (Via, VIA_TYPE);
+  draw_pcb_pv (Via, true);
   if (!TEST_FLAG (HOLEFLAG, Via) && TEST_FLAG (DISPLAYNAMEFLAG, Via))
-    DrawPinOrViaNameLowLevel (Via);
+    draw_pcb_pv_name (Via);
 }
 
 /* ---------------------------------------------------------------------------
